src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -115,23 +115,45 @@ class EdaCredentialCreateSerializer(serializers.ModelSerializer):
     inputs = serializers.JSONField()
 
     def validate(self, data):
-        credential_type_id = data.get("credential_type_id")
-        if credential_type_id:
-            credential_type = models.CredentialType.objects.get(
-                id=credential_type_id
-            )
-        else:
-            # for update
-            credential_type = self.instance.credential_type
+        credential_type = models.CredentialType.objects.get(
+            id=data.get("credential_type_id")
+        )
```

Comment:
Is the field validator executed before this function? I wonder if this raises DoesNotExist if the id does not exist.

Reply:
I'm positive that field validators are executed first before the `validate` function and since `credential_type_id` is required, we can assume we'll always have a value for it.